docs/source/speed.mdx

```diff
@@ -0,0 +1,83 @@
+<!-- DISABLE-FRONTMATTER-SECTIONS -->
+
+# Speed Comparison
+
+[[open-in-colab]]
+
+`Safetensors` is really fast. Let's compare it against `PyTorch` by loading [gpt2](https://huggingface.co/gpt2) weights. To run the [GPU benchmark](#gpu-benchmark), make sure your machine has GPU or you have selected `GPU runtime` if you are using Google Colab.
+
+Before you begin, make sure you have all the necessary libraries installed:
+
+```bash
+!pip install safetensors huggingface_hub torch
+```
+
+Let's start by importing all the packages that will be used:
+
+```py
+>>> import os
+>>> import datetime
+>>> from huggingface_hub import hf_hub_download
+>>> from safetensors.torch import load_file
+>>> import torch
+```
+
+Download safetensors & torch weights for gpt2:
+
+```py
+>>> sf_filename = hf_hub_download("gpt2", filename="model.safetensors")
+>>> pt_filename = hf_hub_download("gpt2", filename="pytorch_model.bin")
+```
+
+### CPU benchmark
+
+```py
+>>> # Make sure the experimental gpu loading feature is off since we are testing CPU here
+>>> if "SAFETENSORS_FAST_GPU" in os.environ: del os.environ["SAFETENSORS_FAST_GPU"]
+
+>>> device = "cpu"
+
+>>> start_st = datetime.datetime.now()
+>>> weights = load_file(sf_filename, device=device)
+>>> load_time_st = datetime.datetime.now() - start_st
+>>> print(f"Loaded safetensors {load_time_st}")
+
+>>> start_pt = datetime.datetime.now()
+>>> weights = torch.load(pt_filename, map_location=device)
+>>> load_time_pt = datetime.datetime.now() - start_pt
+>>> print(f"Loaded pytorch {load_time_pt}")
+
+>>> print(f"on CPU, safetensors is faster than pytorch by: {load_time_pt/load_time_st:.1f} X")
+Loaded safetensors 0:00:00.004015
```

Comment:
As much as we have to do clickbaity stuff on social media to get visibility unfortunately, I'd rather be conservative here, and more importantly explain WHY.

So we can keep the values, but we have to explain a little on what hardware it was taken, and that it works because it's relying on disk cache.

Probably something after the code like:

"'"
This speedup is due to the fact that this library avoids unnecessary copies by mapping the file directly. It is actually possible to do on [pure pytorch](https://gist.github.com/Narsil/3edeec2669a5e94e4707aa0f901d2282).

The current show speedup was gotten on X (os) Y (CPU) Z(disk type).
"""

Wdyt ?

Reply:
https://github.com/huggingface/safetensors/pull/80/commits/0ee6bd225dd0ed1adbde2f26d0338095a0cb75a8

I'm not sure how to find disk info on google colab, which I;m using

Reply:
Then it's OK :)